### DISCUSSION TITLE
FastAPI app with `separate_input_output_schemas` disabled excludes computed fields on Pydantic models from OpenAPI output

### DISCUSSION THREAD

**Discussion by ldthorne** [fastapi/fastapi#10342] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import pytest
from fastapi import FastAPI
from starlette.testclient import TestClient


@pytest.fixture(name="client")
def get_client():
    app = FastAPI(separate_input_output_schemas=False)

    from pydantic import BaseModel, computed_field

    class Rectangle(BaseModel):
        width: int
        length: int

        @computed_field
        @property
        def area(self) -> int:
            return self.width * self.length

    @app.get("/")
    def read_root() -> Rectangle:
        return Rectangle(width=3, length=4)

    client = TestClient(app)
    return client


def test_get(client: TestClient):
    response = client.get("/openapi.json")
    data = response.json()

    rectangle_model = data["components"]["schemas"]["Rectangle"]["properties"]

    assert "area" in rectangle_model.keys()
```

### Description

I expect that even with `separate_input_output_schemas` set to `False` the OpenAPI schema includes computed fields declared on a Pydantic model

### Operating System

macOS

### FastAPI Version

0.103.1

### Pydantic Version

2.4.1

### Python Version

3.8.16

### Additional Context

I think it may come from [here](https://github.com/tiangolo/fastapi/blob/073e7fc950f1af24634b304afbb34bf212756dff/fastapi/_compat.py#L186)

**Comment by krugi** [comment]
@ldthorne I encountered the same issue. Changing `"validation"` to `"serialization"` seems to resolve the problem. @tiangolo, I'd appreciate your input on this. If you agree, I can open a PR to implement this fix, though I'm unsure if there are other implications. 🙏

**Comment by vgrafe** [comment]
I just hit the same issue too. @krugi can you explain the solution you mentioned?

**Comment by vgrafe** [comment, marked as answer]
There's no way to directly create new issues, so I opened a PR here adding a test that should help maintainers: https://github.com/fastapi/fastapi/pull/13207

**Reply by tiangolo** [reply]
Thanks @vgrafe! I created the issue for completeness here: https://github.com/fastapi/fastapi/issues/14431

Thanks for the PR! It's merged now.

It will be available in FastAPI 0.123.4, released in the next hours. :tada:

**Reply by vgrafe** [reply]
Awesome, I really appreciate your help on this!!
